### PR TITLE
Use pose_broadcaster to publish the TCP pose

### PR DESCRIPTION
**Pull request by fmauch**
This uses the PoseBroadcaster currently written by @RobertWilbrandt. As he wants to contribute that to ros2_controllers, this will stay in a draft state until it is merged upstream.

### CODE REVIEW

File: Universal_Robots_ROS2_Driver.jazzy.repos [hunk 1]
```diff
@@ -5,8 +5,8 @@ repositories:
     version: master
   Universal_Robots_ROS2_Description:
     type: git
-    url: https://github.com/UniversalRobots/Universal_Robots_ROS2_Description.git
-    version: rolling
+    url: https://github.com/fmauch/Universal_Robots_ROS2_Description.git
+    version: pose_broadcaster
```
Comment (fmauch) [inline Universal_Robots_ROS2_Driver.jazzy.repos:9]:
```suggestion
    url: https://github.com/UniversalRobots/Universal_Robots_ROS2_Description.git
    version: rolling
```

File: Universal_Robots_ROS2_Driver.jazzy.repos [hunk 2]
```diff
@@ -17,8 +17,8 @@ repositories:
     version: master
   ros2_controllers:
     type: git
-    url: https://github.com/ros-controls/ros2_controllers
-    version: master
+    url: https://github.com/RobertWilbrandt/ros2_controllers
+    version: pose_broadcaster
```
Comment (fmauch) [inline Universal_Robots_ROS2_Driver.jazzy.repos:21]:
```suggestion
    url: https://github.com/ros-controls/ros2_controllers
    version: master
```

File: Universal_Robots_ROS2_Driver.rolling.repos [hunk 3]
```diff
@@ -5,8 +5,8 @@ repositories:
     version: master
   Universal_Robots_ROS2_Description:
     type: git
-    url: https://github.com/UniversalRobots/Universal_Robots_ROS2_Description.git
-    version: rolling
+    url: https://github.com/fmauch/Universal_Robots_ROS2_Description.git
+    version: pose_broadcaster
```
Comment (fmauch) [inline Universal_Robots_ROS2_Driver.rolling.repos:9]:
```suggestion
    url: https://github.com/UniversalRobots/Universal_Robots_ROS2_Description.git
    version: rolling
```

File: Universal_Robots_ROS2_Driver.rolling.repos [hunk 4]
```diff
@@ -17,8 +17,8 @@ repositories:
     version: master
   ros2_controllers:
     type: git
-    url: https://github.com/ros-controls/ros2_controllers
-    version: master
+    url: https://github.com/RobertWilbrandt/ros2_controllers
+    version: pose_broadcaster
```
Comment (fmauch) [inline Universal_Robots_ROS2_Driver.rolling.repos:21]:
```suggestion
    url: https://github.com/ros-controls/ros2_controllers
    version: master
```

**Comment by gavanderhoorn** [comment]
High-level question: would it perhaps be better to have the hardware interface offer a quaternion instead of an Euler angle triplet?

That would seem cleaner to me, in the sense of it being more of an `N-to-1` mathematical abstraction.

While they aren't necessarily very human friendly, quaternions have definite benefits mathematically, and making it the responsibility of the hardware interface/driver to convert whatever internal orientation representation it uses to this 'intermediary representation' would seem to make sense.

**Comment by fmauch** [comment]
> High-level question: would it perhaps be better to have the hardware interface offer a quaternion instead of an Euler angle triplet?
> 
> That would seem cleaner to me, in the sense of it being more of an `N-to-1` mathematical abstraction.
> 
> While they aren't necessarily very human friendly, quaternions have definite benefits mathematically, and making it the responsibility of the hardware interface/driver to convert whatever internal orientation representation it uses to this 'intermediary representation' would seem to make sense.

I was discussing that with @RobertWilbrandt, as well and I would agree. But I think that's a question to discuss once the PR on ros2_controllers is there.

**Comment by gavanderhoorn** [comment]
Oh, just noticed #856 actually does it that way.

**Comment by RobertWilbrandt** [comment]
I don't have any strong opinions on this - Going through [REP 103](https://www.ros.org/reps/rep-0103.html#rotation-representation) i went through the options and preferred RPY because the values cannot represent any invalid state (while i would expect that a quaternion should be normalized). From an application point of view i agree, so i changed the representation in pose_broadcaster now.

**Comment by fmauch** [comment]
Binary builds fail as the necessary changes in ros2_control have just been merged into master.

**Review by VinDp — APPROVED** [review]
Looks good to me and works as inteded.

**Pull request merged**